src/study/system-model/model.cpp

```diff
@@ -136,4 +136,23 @@ ModelBuilder& ModelBuilder::withConstraints(std::vector<Constraint>&& constraint
     return *this;
 }
 
+/**
+ * \brief Sets the ports of the model.
+ *
+ * \param ports A vector of Port objects to set.
+ * \return Reference to the ModelBuilder object.
+ *
+ * inputs it not garanteed to be valid after the call
+ */
+ModelBuilder& ModelBuilder::withPortFieldDefinitions(
+  std::vector<PortFieldDefinition>&& portFieldDefinitions)
+{
+    std::transform(portFieldDefinitions.begin(),
+                   portFieldDefinitions.end(),
+                   std::inserter(model_.portFieldDefinitions_, model_.portFieldDefinitions_.end()),
+                   [](/*Non const to prevent copy*/ PortFieldDefinition& pfd)
+                   { return std::make_pair(pfd.getPort().Id(), std::move(pfd)); });
```

Comment:
The order of evaluation for arguments is unspecified, if the 2nd statement is executed before the 1st, then you might read invalid (moved) memory.

Example https://gist.github.com/flomnes/5b62b714e5f90ce625954a32f5e3d310

Reply:
```suggestion
                   {
                   	auto id = pfd.getPort().Id();
                   	return std::make_pair(id, std::move(pfd)); 
                   });
```

Reply:
done